tac/agents/participant/v1/agent.py

```diff
@@ -145,6 +145,7 @@ def start(self, rejoin: bool = False) -> None:
             self.oef_handler.rejoin = rejoin
             super().start()
             self.oef_handler.rejoin = False
```

Comment:
hmm ...not sure why we are calling stop in start?

Reply:
I agree it shouldn't be like that. However, somehow the main loop was exited without calling `stop()` but setting the liveness manually, hence bypassing the call to `self.stop()` and hence also the call to `self.teardown()`